code/modules/food_and_drinks/food/snacks_pizza.dm

```diff
@@ -139,6 +139,31 @@
 	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "meat" = 1)
 	foodtype = GRAIN | VEGETABLES | DAIRY
 
+/obj/item/reagent_containers/food/snacks/pizza/pineapple
+	name = "pineapple pizza"
+	desc = "The pizza equivalent of Einstein's riddle."
+	icon_state = "pineapplepizza"
+	slice_path = /obj/item/reagent_containers/food/snacks/pizzaslice/pineapple
+	bonus_reagents = list("nutriment" = 6, "vitamin" = 6)
+	tastes = list("crust" = 1, "tomato" = 1, "cheese" = 1, "pineapples" = 2)
+	foodtype = GRAIN | VEGETABLES | DAIRY
+
+/obj/item/reagent_containers/food/snacks/pizza/pineapple/Initialize()
+	. = ..()
+	if(prob(50))
+		foodtype |= GROSS //either you love pineapple pizza, or you hate pineapple pizza.
```

Comment:
this isn't how flavour works, just remove this

if you hate pineapple pizza fucking roleplay it out

Reply:
is a single joke about the controversy of pineapple pizza really terrible to implement in code

Reply:
yes, it is terrible, because people don't get a choice about a character quirk.